### PR TITLE
Exercise 1

### PR DESCRIPTION
**Pull request by MichaelHoang001**

### CODE REVIEW

File: core/src/main/scala/design/hamu/core/command/Command.scala [hunk 1]
```diff
@@ -8,5 +8,6 @@ object Command {
 
   def apply(string: String): Command = string match {
     case "greet" => GreetCommand
+	case "fib" => FibCommand
```
Comment (vangogh500) [inline core/src/main/scala/design/hamu/core/command/Command.scala]: CI pipeline is yelling about code formatting. You can fix by running `scalafmtAll` in sbt. :+1:

File: core/src/main/scala/design/hamu/core/command/FibCommand.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+object FibCommand extends Command {
+
+	// catch parameter, convert to int, pass to fib()
+	def run(ls: List[String]): String = ls match {
+		case (head:String) :: (tail: List[String]) => {
+			fib(toInt(head)).toString
+		}
+	}
+
+	// helper function i stole off the internet
```
Comment (vangogh500) [inline core/src/main/scala/design/hamu/core/command/FibCommand.scala]: 😆

Comment (vangogh500) [reply core/src/main/scala/design/hamu/core/command/FibCommand.scala]: You could have also just used "1".toInt

File: core/src/main/scala/design/hamu/core/command/FibCommand.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	def toInt(s: String): Int = {
+		try {
+			s.toInt
+		} catch {
+			case e: Exception => 0
+		}
+	}
+
+	def fib(n: Int): Int = n match {
```
Comment (vangogh500) [inline core/src/main/scala/design/hamu/core/command/FibCommand.scala]: Nice use of match statement :+1:

File: core/src/test/scala/design/hamu/core/command/FibCommandSpec.scala [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package design.hamu.core.command
+
+import design.hamu.core.command.FibCommand
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class FibCommandSpec extends AnyWordSpec with Matchers {
+  "Fib command" should {
+    "generate the fibbonaci numbers" in {
```
Comment (vangogh500) [inline core/src/test/scala/design/hamu/core/command/FibCommandSpec.scala:9]: I'd have 2 test cases 1 for positive and 1 for negative since you have some logic with your implementation that forks the logic between the 2.

Comment (vangogh500) [reply core/src/test/scala/design/hamu/core/command/FibCommandSpec.scala:9]: Also best practice is to write a unit test for each function. Where each unit test covers the edge cases specific to that function. A little trivial here since it's so small, but I'd include it to get in the habit.

File: core/src/main/scala/design/hamu/core/command/FibCommand.scala [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+		case 0 => 0
+		case 1 => 1
+		case _ => (fib(n-1) + fib(n-2))
```
Comment (wruoting) [inline core/src/main/scala/design/hamu/core/command/FibCommand.scala]: Btw you couldve made this tail recursion and i think it'll save some more memory

**Review by vangogh500 — APPROVED** [review]
Looks good :+1:
You may have to do one last merge with master with conflict resolution. But after feel free to squash and merge :)
